doc/source/administrator/authentication.md

```diff
@@ -0,0 +1,427 @@
+9.  Type "consent screen" in the search field at the top and click to
+    access the OAuth consent screen. Here you will customize what your
+    users see when they login to your JupyterHub instance for the first
+    time. Click Save when you are done.
+10. In your helm chart, create a stanza that contains these OAuth
+    fields:
+
+```yaml
+hub:
+  config:
+    OAuthenticator:
+      client_id: your-client-id.apps.googleusercontent.com
+      client_secret: your-client-secret
+      oauth_callback_url: https://your-jupyterhub-domain/hub/oauth_callback
+    GoogleOAuthenticator:
+      hosted_domain:
+        - your-university.edu
+      login_service: Your university
+    JupyterHub:
+      authenticator_class: oauthenticator.GoogleOAuthenticator
+```
+
+The `oauth_callback_url` key is set to the authorized redirect URI you
+specified earlier. Set `hosted_domain` to your institution's domain
+name. The value of `login_service` is a descriptive term for your
+institution that reminds your users which account they are using to
+login.
+
+#### Authenticating with LDAP
+
+JupyterHub supports LDAP and Active Directory authentication. Read the
+[ldapauthenticator](https://github.com/jupyterhub/ldapauthenticator)
+documentation for a full explanation of the available parameters. The
+full mapping between parameters set in `values.yaml` and
+`ldapauthenticator` parameter names can be found in
+[jupyterhub_config.py](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/jupyterhub/files/hub/jupyterhub_config.py#L353).
+
+#### Example LDAP Configuration
+
+`server_address` and `bind_dn_template` are
+required. Other fields are optional.
+
+```yaml
+hub:
+  config:
+    JupyterHub:
+      authenticator_class: ldapauthenticator.LDAPAuthenticator
+    LDAPAuthenticator:
+      bind_dn_template:
+        - cn={username},ou=edir,ou=people,ou=EXAMPLE-UNIT,o=EXAMPLE
+      server_address: ldap.EXAMPLE.org
+```
+
+#### Example Active Directory Configuration
+
+This example is equivalent to that given in the [ldapauthenticator
+README](https://github.com/jupyterhub/ldapauthenticator/blob/master/README.md).
+
+```yaml
+hub:
+  config:
+    JupyterHub:
+      authenticator_class: ldapauthenticator.LDAPAuthenticator
+    LDAPAuthenticator:
+      allowed_groups:
+        - cn=researcher,ou=groups,dc=wikimedia,dc=org
+        - cn=operations,ou=groups,dc=wikimedia,dc=org
+      bind_dn_template:
+        - uid={username},ou=people,dc=wikimedia,dc=org
+        - uid={username},ou=developers,dc=wikimedia,dc=org
+      escape_userdn: false
+      lookup_dn: true
+      lookup_dn_search_filter: ({login_attr}={login})
+      lookup_dn_search_password: secret
+      lookup_dn_search_user: ldap_search_user_technical_account
+      lookup_dn_user_dn_attribute: cn
+      server_address: ad.EXAMPLE.org
+      user_attribute: sAMAccountName
+      user_search_base: ou=people,dc=wikimedia,dc=org
+```
+
+#### Example Auth0 Configuration
```

Comment:
I'm a bit confused, there's an earlier ` #### Auth0` section that uses `oauthenticator.auth0.Auth0OAuthenticator` but this is different? Is to illustrate OpenID (and Auth0 just happens to be an example OpenID provider)?

Reply:
I'm also confused, I decided to not care about this difference as part of this PR.

Reply:
I went ahead and updated the entire authentication page where I located two examples under a GenericOAuthenticator section, where I took Auth0 and KeyCloak examples against OpenID connect.